src/modules/test/RecipeInvoker.cpp

```diff
@@ -82,11 +80,10 @@ void RecipeInvoker::TestBody()
                 JSON_Value *recipe_payload = json_parse_string(m_recipe.m_payload.c_str());
                 JSON_Value *returned_payload = json_parse_string(payload);
 
-                std::string recipe_payload_str(payload, payloadSize);
                 EXPECT_NE(nullptr, recipe_payload) << "Failed to parse recipe payload" << std::endl << "JSON: " << m_recipe.m_payload.c_str() << std::endl;
-                EXPECT_NE(nullptr, returned_payload) << "Failed to parse returned payload" << std::endl << "JSON: " << recipe_payload_str.c_str() << std::endl;
+                EXPECT_NE(nullptr, returned_payload) << "Failed to parse returned payload" << std::endl << "JSON: " << payloadString.c_str() << std::endl;
                 EXPECT_EQ(json_value_get_type(returned_payload), json_value_get_type(recipe_payload)) << "Non matching payload types. Recipe payload: " << json_value_get_type(recipe_payload) << ", returned payload: " << json_value_get_type(returned_payload) << std::endl;
-                EXPECT_EQ(json_value_equals(recipe_payload, returned_payload), 1) << "Non matching recipe payload" << std::endl << "Recipe   payload: " << m_recipe.m_payload << std::endl << "Returned payload: " << recipe_payload_str.c_str() << std::endl;
+                EXPECT_STRCASEEQ(m_recipe.m_payload.c_str(), payloadString.c_str()) << "Non matching recipe payload" << std::endl << "Recipe   payload: " << m_recipe.m_payload << std::endl << "Returned payload: " << payloadString.c_str() << std::endl;
```

Comment:
Will this also work for objects? What if two objects are equal but serialized differently (should be equal but different serialization means they will not be equal)?

Reply:
It's all strings. It'll work for object payloads as long as expected will be in same order and format with actual

Reply:
yeah, these will be known strings so it will not be an issue

Reply:
In general I agree that a json object shouldn't care about member ordering, but for the recipes, we should note that we expect the object to be ordered just as the module will be serialized. Given that the recipes are authored by the module authors themselves this shouldn't be an issue, the expected json shouldn't be a surprise.

Reply:
Spaces vs. no spaces will break this comparison, even if same order and names

Reply:
These should be compared as JSON values, it will be very difficult to debug and identify failures caused by small serialization differences in large JSON objects (difference of spacing, ordering, etc).

Reply:
Spaces before/after elements in JSON can break this comparison even with same ordering

Reply:
Example:

"{\"Name\" :  \"Configuration\","
    "\"Description\" :  \"Provides functionality to manage OSConfig configuration on device\","....

vs

"{\"Name\": \"Configuration\","
    "\"Description\": \"Provides functionality to manage OSConfig configuration on device\","

Reply:
Also lines with EOLs vs not.